docs: update supported platform information (#2594)
thin-edge/thin-edge.io: docs/src/references/supported-platforms.md

# thin-edge.io platform support

thin-edge.io can be run on any Linux based Operating System, such as; Debian/Ubuntu, Alpine, RHEL, Fedora, Pokey (Yocto) etc.

In addition to a Linux based Operation system, the following requirements must also be fulfilled:

> * At least ~40MB of RAM (see [notes](#memory-usage))

Reviewer: ```suggestion
* At least ~40MB of RAM on the gateway device, ~16MB on a child device (see [notes](#memory-usage))
```

Author: Good suggestion, resolved by 210c25540477ba59e02c590be60674e400841977

* [mosquitto](https://github.com/eclipse/mosquitto) MQTT Broker (see [notes](#recommended-mosquitto-version))

:::tip
If you are looking for new hardware it is highly recommended to choose a CPU which has at least 2 cores, as it provides a much more responsive experience and should provide enough processing headroom for future requirements.
:::

:::note
mosquitto will be installed via the installation script, however it needs to be available from a software repository, however mosquitto is generally available in most popular Linux based operating systems.
:::

## Recommended mosquitto version

Whilst thin-edge.io will work with any mosquitto version, it is recommended to use mosquitto version &gt;=2.0.18 as it includes some important fixes regarding the MQTT bridge functionality.

If you can't use the recommended version, then be aware of the following issues which affect various mosquitto versions in some corner cases:

|Mosquitto Version|Issues|
|-----------------|------|
|2.0.11|https://github.com/eclipse/mosquitto/issues/2604|
|2.0.15|https://github.com/eclipse/mosquitto/issues/2634|

## Supported CPU Architectures

> thin-edge.io is developed in Rust which is an ahead-of-time compiled language. This means that not only does thin-edge.io need to be compiled to produce the executable, that each executable is specific for different CPU Architectures (e.g. x86_64/amd64, aarch64/arm64, armv7, armv6 etc.).

Reviewer: Isn't this common knowledge?

```suggestion
thin-edge.io is developed in Rust which is a compiled language and each executable is specific for different CPU Architectures (e.g. x86_64/amd64, aarch64/arm64, armv7, armv6 etc.).
```

Author: I wouldn't assume this is common knowledge...especially if people are coming from the python world where generally the user does not have to think about the CPU architecture...but I'll reword as per your suggestion.
